core/primitives/src/utils.rs

```diff
@@ -407,19 +406,11 @@ pub fn to_timestamp(time: DateTime<chrono::Utc>) -> u64 {
     time.timestamp_nanos() as u64
 }
 
-/// Compute number of seats per shard for given total number of seats and number of shards.
-pub fn get_num_seats_per_shard(num_shards: NumShards, num_seats: NumSeats) -> Vec<NumSeats> {
-    (0..num_shards)
-        .map(|i| {
-            let remainder = num_seats % num_shards;
-            let num = if i < remainder as u64 {
-                num_seats / num_shards + 1
-            } else {
-                num_seats / num_shards
-            };
-            max(num, 1)
-        })
-        .collect()
+pub fn get_num_seats_per_shard(
+    num_shards: NumShards,
+    num_seats_per_shard: NumSeats,
```

Comment:
this is a little bit tricky -- as you're changing the meaning of the argument (and this method is called in many places).

Reply:
Should be maybe just inline it everywhere? It exists mostly for tests, and seem to create more confusion than any future-proofing.

Reply:
Yeah the tests are a little tricky. I didn't go through all of the tests, but most of them I saw are only using one shard. I'll remove the function and just try inlining it everywhere so I can check all the tests affected.

Reply:
Actually I was being dumb. The current change should be good.